### DISCUSSION TITLE
Is it possible to have a dynamic url prefix?

### DISCUSSION THREAD

**Discussion by jkrumbiegel** [jameslittle230/stork#335] [Q&A] [answered]
Our site has multiple versions of the same docs, and the most recent version gets the url alias `stable`. Now, stork wants the full address to each search result as far as I know, and that includes the version prefix. The drawback of this is, that users start at `/stable/...` and then do a search which leads them to `/vX.Y/...`. I'd rather not have them leave `stable` if possible, so that urls of sites that don't change much which get posted elsewhere don't have the version in the string. I wonder if we can avoid that somehow? I already have a javascript variable exported during the build which tells me what the current `stable` version refers to. Can I make stork use that?

**Comment by jameslittle230** [comment, marked as answer]
Hi @jkrumbiegel! Thanks for writing in.

I think you have two options. You could either change your url_prefix so that it references the `/stable/` urls, or you can use the [`transformResultUrl` JS API](https://stork-search.net/docs/js-ref#transformResultUrl) to rewrite the URLs from some template string to your version string.

If you were to go that second route, it might look something like setting your config's URL prefix to something like:

```toml
[input]
url_prefix = "https://my-cool-docs-site.net/{VERSION}/"
```

and then, when you register Stork on the page:

```js
const currentVersion = "v2.5.1"; // exported from build system
stork.register("search-index", "https://https://my-cool-docs-site.net/index.st", {
   transformResultUrl: (url) => url.replaceAll("{VERSION}", currentVersion)
});
```

(I'm not a compiler, not 100% sure that that's valid code 😃)

With that system in place, the links in the Stork dropdown will be of the form `https://my-cool-docs-site.net/v2.5.1/foobarFactory.html`.

Let me know if that helps or if I'm missing something!

-James

**Reply by jkrumbiegel** [reply]
Great, I think this is exactly what I needed. I didn't know about `transformResultUrl`, yet, must have missed it scrolling through the docs. Thanks!
